app/src/main/java/co/co/nimblehq/showcases/poly/place/SliceProvider.kt

```diff
@@ -0,0 +1,101 @@
+package co.co.nimblehq.showcases.poly.place
+
+import android.content.ContentResolver
+import android.content.Intent
+import android.net.Uri
+import androidx.slice.Slice
+import androidx.slice.SliceProvider
+import androidx.slice.builders.ListBuilder
+import androidx.slice.builders.SliceAction
+
+class SliceProvider : SliceProvider() {
+    /**
+     * Instantiate any required objects. Return true if the provider was successfully created,
+     * false otherwise.
+     */
+    override fun onCreateSliceProvider(): Boolean {
+        return true
+    }
+
+    /**
+     * Converts URL to content URI (i.e. content://co.co.nimblehq.showcases.poly.place...)
+     */
+    override fun onMapIntentToUri(intent: Intent?): Uri {
+        // Note: implementing this is only required if you plan on catching URL requests.
+        // This is an example solution.
+        var uriBuilder: Uri.Builder = Uri.Builder().scheme(ContentResolver.SCHEME_CONTENT)
```

Comment:
I think we can remove the type here since it has already been assigned value to Uri Builder 😄 
```suggestion
        var uriBuilder = Uri.Builder().scheme(ContentResolver.SCHEME_CONTENT)
```

Reply:
@thiennguyen0196 Got it. FYI, It's generated class by AS.

Reply:
@minhnimble @thiennguyen0196 Wait, when I rethink again, what do you think to avoid refactoring since it's the generated class by the AS? I'm not sure in this case. Do we need to re-format every generated class?

Reply:
@minhnimble My point is, this whole class is generated by the AS. So do we need to refactor all generated classes too? I'm just curious. 🤔

Reply:
@lydiasama you got the point when this is a generated class from Android Studio 👏  However since we have already had the convention written in Compass and this class will be included in the final code too, so I think we should refactor the below code a little bit to match our overall code convention 🤔 

Please feel free to correct me if I was wrong 😄

Reply:
@thiennguyen0196 Nope, You're not wrong. I'm just wondering if we need to look into every single generated classes. Let's say you have a thousand lines class being generated, do you need to reformat it every line? 🤔

Reply:
@minhnimble Do we have any convention about this?

Reply:
@lydiasama please correct me if I am wrong but even though it was generated, we will have more updates on this file to handle our custom logic later on, right? 

If yes, then refactoring the code to match our team convention is needed. I think we should only keep when we rely on the code generation solely and we won't perform any custom update on the file 💭

Reply:
@minhnimble To be honest, I don't have time to play around with the Slice Template much. Can we reformat it once we do the real implementation for all this class, please? I'd prefer to do that instead of updating it from generating time. cc @thiennguyen0196

Reply:
> @minhnimble Do we have any convention about this?

@lydiasama I don't think we need it, it should be just common sense LOL 😁

Reply:
> @minhnimble To be honest, I don't have time to play around with the Slice Template much. Can we reformat it once we do the real implementation for all this class, please? I'd prefer to do that instead of updating it from generating time. cc @thiennguyen0196

@lydiasama I have no issue with perform refactoring at a later time. Let's do that at in the subsequent PRs 🚀

Reply:
I agree @lydiasama 👍  just to make sure we need to have a chore task to refactor all the slice template generated class file once we finish all the task 😄 

Then you could mark all the below comments as resolved 🚀

Reply:
@thiennguyen0196 Alright, I'll create a chore issue for that. But I think we would do it together with the `feature` task to implement the Slice Template by the way. 😂

Reply:
@thiennguyen0196 The issue is [here](https://github.com/nimblehq/poly-place/issues/26)